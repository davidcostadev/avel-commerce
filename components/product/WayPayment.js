import React from 'react'
import PropTypes from 'prop-types'
import classname from 'classnames'
import styled from 'styled-components'

const listPayments = [
  // {
  //   id: 13,
  //   name: 'Pagseguro',
  // },
  // {
  //   id: 15,
  //   name: 'Boleto',
  // },
  // {
  //   id: 17,
  //   name: 'Depósito',
  // },
  {
    id: 1,
    name: 'Pagar na Loja',
  },
  {
    id: 2,
    name: 'Cartão de Débido',
  },
  {
    id: 22,
    name: 'Cartão de Crédito até 3x',
  },
]

const Ways = ({ paymentDefault, onClick }) => (
  <div>
    {
      listPayments.map(payment => (
        <Way
          key={payment.id}
          paymentDefault={paymentDefault}
          payment={payment}
          onClick={onClick}
        />
      ))
    }
  </div>
)

Ways.propTypes = {
  paymentDefault: PropTypes.number.isRequired,
  onClick: PropTypes.func.isRequired,
}

const ButtonPayment = styled.button`
  margin-bottom: 30px;

  & + & {
    margin-left: 15px;
  }
`

const Way = ({ paymentDefault, payment, onClick }) => {
  const classPayment = classname(
    'btn',
    'btn-default',
    {
      'btn-primary': paymentDefault === payment.id,
    },
  )
  return (
    <ButtonPayment
      type="button"
      className={classPayment}
      onClick={() => onClick(payment.id)}
    >
      {payment.name}
    </ButtonPayment>
  )
}

Way.propTypes = {
  paymentDefault: PropTypes.number.isRequired,
  payment: PropTypes.shape({
    id: PropTypes.number.isRequired,
    name: PropTypes.string.isRequired,
  }).isRequired,
  onClick: PropTypes.func.isRequired,
}

export default Ways
