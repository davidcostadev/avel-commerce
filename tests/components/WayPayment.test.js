/* eslint no-undef: "off" */
import React from 'react'
import Enzyme, { shallow } from 'enzyme'
import Adapter from 'enzyme-adapter-react-16'
import renderer from 'react-test-renderer'
import WayPayment, { Way } from '../../components/product/WayPayment'

Enzyme.configure({ adapter: new Adapter() })

describe('should', () => {
  it('List way payment', () => {
    const onClick = jest.fn()

    const tree = renderer.create((
      <WayPayment
        paymentDefault={1}
        onClick={onClick}
      />
    )).toJSON()
    expect(tree).toMatchSnapshot()
  })

  it('Item way payment', () => {
    const onClick = jest.fn()
    const payment = {
      id: 22,
      name: 'one',
      factor: 1.5,
    }
    const tree = renderer.create((
      <Way
        paymentDefault={1}
        payment={payment}
        onClick={onClick}
      />
    )).toJSON()
    expect(tree).toMatchSnapshot()
  })

  it('change way payment', () => {
    const paymentItem = {
      id: 22,
      name: 'one',
      factor: 1.5,
    }
    const onClick = (payment) => {
      expect(payment).toEqual(paymentItem)
    }

    const wrapper = shallow((
      <Way
        paymentDefault={1}
        payment={paymentItem}
        onClick={onClick}
      />
    ))

    expect(wrapper.get(0)).toMatchSnapshot()
    wrapper.find('#payment-22').simulate('click')
  })
})
