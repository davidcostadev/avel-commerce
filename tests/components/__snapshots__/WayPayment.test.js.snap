// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should Item way payment 1`] = `
<button
  className="btn btn-default WayPayment__ButtonPayment-b3out0-0 dkxUjX"
  id="payment-22"
  onClick={[Function]}
  type="button"
>
  one
</button>
`;

exports[`should List way payment 1`] = `
<div>
  <button
    className="btn btn-default btn-primary WayPayment__ButtonPayment-b3out0-0 dkxUjX"
    id="payment-1"
    onClick={[Function]}
    type="button"
  >
    Pagar na Loja
  </button>
  <button
    className="btn btn-default WayPayment__ButtonPayment-b3out0-0 dkxUjX"
    id="payment-2"
    onClick={[Function]}
    type="button"
  >
    Cartão de Débido
  </button>
  <button
    className="btn btn-default WayPayment__ButtonPayment-b3out0-0 dkxUjX"
    id="payment-22"
    onClick={[Function]}
    type="button"
  >
    Cartão de Crédito até 3x
  </button>
</div>
`;

exports[`should change way payment 1`] = `
<WayPayment__ButtonPayment
  className="btn btn-default"
  id="payment-22"
  onClick={[Function]}
  type="button"
>
  one
</WayPayment__ButtonPayment>
`;
